ggshield/scan/repo.py

```diff
@@ -73,8 +72,12 @@ def scan_commits_content(
     ignored_detectors: Optional[Set[str]] = None,
     ignore_known_secrets: bool = False,
 ) -> ScanCollection:  # pragma: no cover
+    commit_files_tuple = []
     try:
-        commit_files = list(itertools.chain.from_iterable(c.files for c in commits))
+        for commit in commits:
+            for file in commit.files:
+                commit_files_tuple.append((commit, file))
```

Comment:
I would rename `commit_files_tuple` to `commit_files_tuples`, since its a list of tuples and not a list.

Also, I think you can create it using a comprehension list, like this (untested):

```
commit_files_tuples = [(c, f) for c in commits for f in c.files]
```

Reply:
indeed, thanks !